sdk/azidentity/chained_token_credential_test.go

```diff
@@ -26,13 +26,13 @@ func TestChainedTokenCredential_InstantiateSuccess(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Could not find appropriate environment credentials")
 	}
-	cred, err := NewChainedTokenCredential(secCred, envCred)
+	cred, err := NewChainedTokenCredential(secCred.(azcore.TokenCredential), envCred.(azcore.TokenCredential))
```

Comment:
This is unfortunate.  It means that when callers create a chained token credential they have to perform this type-assertion.  I feel like this is a bit of a step backwards.

Reply:
This makes me wonder if this abstraction is wrong.  Should it be `NewChainedCredential()` instead, that takes multiple `azcore.Credential` instead?

Reply:
I agree, this makes ChainedTokenCredential harder to use and it's definitely not ideal. I would not like to leave it like this. But the abstraction for ChainedTokenCredential is correct in being `azcore.TokenCredential` since it is expecting a chain of only TokenCredentials with the GetToken method.

Reply:
Agreed that `ChainedTokenCredential()` takes a slice of `azcore.TokenCredential`.  With this change I'm suggesting it should be `ChainedCredential()` that takes a slice of `azcore.Credential`.  How does a customer know they need to perform this type assertion?

Another option would be to leave it as `ChainedTokenCredential()` that takes a slice of `azcore.Credential` and perform the type-assertion in the constructor, returning an error if it fails.  I don't like this though as you lose the type safety.

I still think the best option is to return concrete types from the credential constructors as it allows them to be covariant.